File: Arrays_Hashing/LongestConsecutive/LongestConsecutive.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
    val mySet = HashSet<Int>()

    for (n in nums) {
        mySet.add(n)
    }

    var longest = 0

    for (n in mySet) {
        if ((n - 1) !in mySet) {
          
            var current = n
            var length = 1

            while ((current + 1) in mySet) {
                current += 1
                length += 1
            }

            longest = maxOf(longest, length)
        }
    }

    return longest
}
}

File: Arrays_Hashing/LongestConsecutive/LongestConsecutive.swift
func longestConsecutive(_ nums: [Int]) -> Int {
    var mySet = Set<Int>()
    
    for i in 0..<nums.count {
        mySet.insert(nums[i])
    }
    
    var longest = 0  
    
    for n in mySet {
        if !mySet.contains(n-1) {
            var current = n
            var length = 1
            while mySet.contains(current + 1) {
                current += 1
                length += 1
            }
            
            longest = max(longest, length)
            
        }
    }
    
    return longest    
}

